tests/rendering_test.py

```diff
@@ -59,29 +62,41 @@ def test_render_blank_markup_with_custom_json_encoder():
     ''')
 
 
-def test_render_blank_markup_with_error():
+@pytest.mark.parametrize('error, errorMarkup', [
+    (
+        HypernovaError('Error', 'Error msg', ['1: Error', '2: stack']),
+        '["Error", "Error msg", ["1: Error", "2: stack"]]'
+    ),
+    (None, 'undefined'),
+])
+def test_render_blank_markup_when_throw_client_error_true(error, errorMarkup):
```

Comment:
mild nit that doing this parametrised makes it slightly more difficult to tell what the expected inputs/outputs are vs. a more verbose but explicit set of tests that hardcode those values directly

the most important thing is that we test the variations, which you seem to have done, so nbd - just mentioning it if we ever expand this functionality :)